src/panels/lovelace/cards/hui-area-card.ts

```diff
@@ -0,0 +1,413 @@
+      tap_action: { action: "more-info" },
+      hold_action: { action: "more-info" },
+      ...entityConfig,
+    })) as EntitiesCardEntityConfig[];
+
+    return html`
+      <ha-card>
+        <img src=${this.hass.hassUrl(this._config.image)} />
+        <div class="container">
+          <div class="sensors">
+            ${dialogEntities.map((entityConf) => {
+              const stateObj = this.hass!.states[entityConf.entity];
+              return html`
+                <span>
+                  <ha-icon
+                    .config=${entityConf}
+                    .icon=${stateIcon(stateObj)}
+                    .actionHandler=${actionHandler({
+                      hasHold: hasAction(entityConf.hold_action),
+                    })}
+                    @action=${this._handleAction}
+                  ></ha-icon>
+                  ${computeStateDisplay(
+                    this.hass!.localize,
+                    stateObj,
+                    this.hass!.locale
+                  )}
+                </span>
+              `;
+            })}
+          </div>
+          <div class="name">${area!.name}</div>
+          <div class="buttons">
+            ${toggleEntities.map((entityConf) => {
+              const stateObj = this.hass!.states[entityConf.entity];
+              return html`
+                <ha-icon-button
+                  .config=${entityConf}
+                  .icon=${stateIcon(stateObj)}
+                  class=${classMap({
+                    off: STATES_OFF.includes(stateObj.state),
+                  })}
+                  .actionHandler=${actionHandler({
+                    hasHold: hasAction(entityConf.hold_action),
+                  })}
+                  @action=${this._handleAction}
+                ></ha-icon-button>
+              `;
+            })}
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  public willUpdate(changedProps: PropertyValues) {
+    if (
+      (!this.hass ||
+        !this._config?.area ||
+        !this._areas ||
+        !this._entities ||
+        !this._devices) &&
+      !changedProps.has("_entities") &&
+      !changedProps.has("_devices")
+    ) {
+      return;
+    }
+
+    this._entitiesDialog = [];
+    this._entitiesToggle = [];
+
+    const { entities } = this._memberships(
+      this._config!.area,
+      this._devices!,
+      this._entities!
+    );
+
+    if (entities.length === 0) {
+      return;
+    }
+
+    entities.forEach((entity) => {
+      const domain = computeDomain(entity.entity_id);
+      const isToggle = DOMAINS_TOGGLE.has(domain);
+      const stateObj = this.hass!.states[entity.entity_id];
+
+      if (isToggle && this._entitiesToggle!.length < 3) {
+        this._entitiesToggle!.push(entity.entity_id);
+      } else if (
+        !isToggle &&
+        stateObj.attributes.device_class &&
+        AREA_SENSOR_CLASSES.includes(stateObj.attributes.device_class) &&
+        this._entitiesDialog!.length < 3
```

Comment:
What are you thoughts here? Limiting to the first 3... Should we sort them in a better way? I know this will get some push back

Reply:
I would sort them temperature, humidity, motion?

Reply:
Same with the Togglable ones as well

Light, Switch, etc